pkg/utils/metatopic/metatopic.go

```diff
@@ -0,0 +1,104 @@
+package metatopic
+
+import (
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/oklog/ulid/v2"
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+// Constants useful for creating ensign Events.
+const (
+	Mimetype      = "application/msgpack"
+	SchemaName    = "metatopic.TopicUpdate"
+	SchemaVersion = "1.0.0"
+)
+
+// This is the top-level event type that is sent on the metatopic topic. It describes an
+// update to a specific topic and provides information about the topic and how it was
+// modified, and, if available, who it was modified by.
+type TopicUpdate struct {
+	UpdateType TopicUpdateType `msgpack:"update_type"`
+	OrgID      ulid.ULID       `msgpack:"org_id,omitempty"`
+	ProjectID  ulid.ULID       `msgpack:"project_id"`
+	TopicID    ulid.ULID       `msgpack:"topic_id"`
+	ClientID   string          `msgpack:"client_id"`
+	Topic      *Topic          `msgpack:"topic,omitempty"`
+}
+
+// A non-protocol buffer representation of the Topic. In the Topic Update struct it
+// represents the modified topic (e.g. the current version of the topic).
+// TODO: add placements and types to this struct.
+type Topic struct {
```

Comment:
Do we need to add data storage and online publishers/subscribers? Or were we thinking those would be separate events?

Reply:
Good point - I'll add those here!